Unable to build Windows class libraries with dotnet build

### Description

We added support for dotnet build, but it seems that libraries still have some issues.

Partial PR: https://github.com/dotnet/maui/pull/5875
Maui devops issue: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1520610
OS devops issue: https://microsoft.visualstudio.com/OS/_workitems/edit/38897015

### Steps to Reproduce

1. Create a new winui class library
2. dotnet build
3. Observe the error saying it can't load the Pri tasks

```
D:\a\_work\_temp\templatesTest\packages\microsoft.windowsappsdk\1.0.2\buildTransitive\MrtCore.PriGen.targets(891,5):
    error MSB4062: The "Microsoft.Build.Packaging.Pri.Tasks.ExpandPriContent" task could not be loaded from the 
    assembly D:\a\_work\1\s\bin\dotnet\sdk\6.0.300-preview.22204.3\\Microsoft\VisualStudio\v17.0\AppxPackage\\Microsoft.Build.Packaging.Pri.Tasks.dll. 
    Could not load file or assembly 'D:\a\_work\1\s\bin\dotnet\sdk\6.0.300-preview.22204.3\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.Build.Packaging.Pri.Tasks.dll'. 
    The system cannot find the path specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are
    available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.
    [D:\a\_work\_temp\templatesTestmauilibTrue\templatesTestmauilibTrue.csproj]
 ```

### Version with bug

Release Candidate 1

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

All

### Did you find any workaround?

Use msbuild :(

### Relevant log output

_No response_

--- a/.nuspec/Microsoft.Maui.Controls.SingleProject.props
+++ b/.nuspec/Microsoft.Maui.Controls.SingleProject.props
@@ -2,7 +2,7 @@
 
   <PropertyGroup Condition=" '$(SingleProject)' == 'true' and '$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == 'windows' ">
     <OutputType Condition="'$(OutputType)' == 'Exe'">WinExe</OutputType>
-    <EnablePreviewMsixTooling Condition="'$(EnablePreviewMsixTooling)' == '' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe')">true</EnablePreviewMsixTooling>
+    <EnablePreviewMsixTooling Condition="'$(EnablePreviewMsixTooling)' == ''">true</EnablePreviewMsixTooling>
     <WindowsPackageType Condition=" '$(WindowsPackageType)' == '' and '$(EnablePreviewMsixTooling)' == 'true' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe') ">MSIX</WindowsPackageType>
     <WinUISDKReferences Condition=" '$(WinUISDKReferences)' == '' and '$(EnablePreviewMsixTooling)' == 'true' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe') ">false</WinUISDKReferences>
     <_SingleProjectRIDRequired Condition="'$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe'">true</_SingleProjectRIDRequired>

--- a/src/Templates/src/templates/maui-lib/MauiLib1.csproj
+++ b/src/Templates/src/templates/maui-lib/MauiLib1.csproj
@@ -2,7 +2,7 @@
 
 	<PropertyGroup>
 		<TargetFrameworks>DOTNET_TFM;DOTNET_TFM-android;DOTNET_TFM-ios;DOTNET_TFM-maccatalyst</TargetFrameworks>
-		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows')) and '$(MSBuildRuntimeType)' == 'Full'">$(TargetFrameworks);DOTNET_TFM-windows10.0.19041.0</TargetFrameworks>
+		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);DOTNET_TFM-windows10.0.19041.0</TargetFrameworks>
 		<RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">MauiLib1</RootNamespace>
 		<UseMaui>true</UseMaui>
 		<SingleProject>true</SingleProject>

